### PR TITLE
Groups and Such

### PR DESCRIPTION
**Pull request by dmfj**
- Begin work on groups.
- Support for multiple roles. (no longer just event organizers)
- Minor optimizations.
- Separate some components to other pages.
- Events now belong to a group.
- New users will be added to the global TU20 group automatically.

- [ ] Production DB work needed before merging.

### CODE REVIEW

**Comment by prathamthukral** [comment]
Lgtm

File: app/Http/Middleware/VerifyKioskStatus.php [hunk 1]
```diff
@@ -15,7 +15,7 @@ class VerifyOrganizerStatus
      */
     public function handle($request, Closure $next)
     {
-        if ($request->user() && $request->user()->is_organizer) {
+        if ($request->user() && $request->user()->role != 'tu20') {
```
Comment (Diophontine) [inline app/Http/Middleware/VerifyKioskStatus.php]: should make this a constant

File: database/migrations/2017_09_06_225329_change_is_organizer_to_role.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('users', function (Blueprint $table) {
+            $table->dropColumn('is_organizer');
```
Comment (Diophontine) [inline database/migrations/2017_09_06_225329_change_is_organizer_to_role.php]: constant?

File: database/migrations/2017_09_06_225329_change_is_organizer_to_role.php [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+    /**
+     * Reverse the migrations.
+     *
+     * @return void
+     */
+    public function down()
+    {
+        Schema::table('users', function (Blueprint $table) {
+            $table->boolean('is_organizer')->default(false);
```
Comment (Diophontine) [inline database/migrations/2017_09_06_225329_change_is_organizer_to_role.php]: constant?

File: .travis.yml [hunk 4]
```diff
@@ -16,6 +16,6 @@ script:
 notifications:
```
Comment (Diophontine) [inline .travis.yml]: Add some unit tests for next pull request

File: database/migrations/2017_09_06_225329_change_is_organizer_to_role.php (hunk 3 above)
Comment (dmfj) [reply database/migrations/2017_09_06_225329_change_is_organizer_to_role.php]: This is the inverse operation of the migration (reset back to the original state)

File: .travis.yml (hunk 4 above)
Comment (dmfj) [reply .travis.yml]: Will do. 👍

File: app/Http/Middleware/VerifyKioskStatus.php (hunk 1 above)
Comment (dmfj) [reply app/Http/Middleware/VerifyKioskStatus.php]: That makes sense, yeah.

File: database/migrations/2017_09_06_225329_change_is_organizer_to_role.php (hunk 2 above)
Comment (dmfj) [reply database/migrations/2017_09_06_225329_change_is_organizer_to_role.php]: Nope, not here.

**Pull request merged**